### PR TITLE
Added rainbow border button

### LINKED ISSUE #882: [BUTTON IDEA]: Rainbow Border Button
**1. Name of button**: Rainbow Border

**2. Description**: a black background button with a radio border

**3. Button Type (Animated, Special, etc...)**: Special

**4. Will you work on it?**:No, please comment on this to be assigned

This button is similar to the button you see on codepen

### PR DESCRIPTION
**Pull request by Soumi150**
Fixes #882

### CODE REVIEW

**Review by shahednasser — COMMENTED** [review]
Thank you for your work @Soumi150 it looks great but on hover it does nothing. Can you fix it please?

**Comment by Soumi150** [comment]
I have added animation but it is not working on hover. Can you please look into the code and guide me a little. Thank you.

File: src/components/special/_rainbow-border.less [hunk 1]
```diff
@@ -0,0 +1,80 @@
+@rainbowBackgroundColor: #000;
+@rainbowBorderColor: #fff;
+.rainbow-border-btn {
+  position: relative;
+  background-color: @rainbowBackgroundColor;
+  color: @rainbowBorderColor;
+  border: 5px solid;
```
Comment (shahednasser) [inline src/components/special/_rainbow-border.less:7]: Remove this

File: src/components/special/_rainbow-border.less [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  border-image-slice: 1;
```
Comment (shahednasser) [inline src/components/special/_rainbow-border.less:8]: Remove this

File: src/components/special/_rainbow-border.less [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  border-image-source: linear-gradient(
```
Comment (shahednasser) [inline src/components/special/_rainbow-border.less:9]: Remove this

File: src/components/special/_rainbow-border.less [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    #fad648,
+    #a767e5,
+    #12bcfe,
+    #44ce7b
+  );
+  display: inline;
+  overflow: hidden;
+
+  &:hover {
+    animation: colorwave 1.6s infinite;
```
Comment (shahednasser) [inline src/components/special/_rainbow-border.less:21]: add ease

File: src/components/special/_rainbow-border.less [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  }
+}
+
+@keyframes colorwave {
+  0% {
+    border-image-source: linear-gradient(
```
Comment (shahednasser) [inline src/components/special/_rainbow-border.less:27]: change this to 
```
background-position: 0 0, 0% 50%;
```

File: src/components/special/_rainbow-border.less [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      114deg,
+      #4fcf70,
+      #fad648,
+      #a767e5,
+      #12bcfe,
+      #44ce7b
+    );
+  }
+
+  25% {
```
Comment (shahednasser) [inline src/components/special/_rainbow-border.less:37]: Remove this

File: src/components/special/_rainbow-border.less [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      #44ce7b,
+      #4fcf70,
+      #fad648,
+      #a767e5,
+      #12bcfe
+    );
+  }
+
+  50% {
+    border-image-source: linear-gradient(
```
Comment (shahednasser) [inline src/components/special/_rainbow-border.less:49]: Change this to 
```
    background-position: 0 0, 100% 50%;
```

File: src/components/special/_rainbow-border.less [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      134deg,
+      #12bcfe,
+      #44ce7b,
+      #4fcf70,
+      #fad648,
+      #a767e5
+    );
+  }
+
+  75% {
```
Comment (shahednasser) [inline src/components/special/_rainbow-border.less:59]: Remove this

File: src/components/special/_rainbow-border.less [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+      #a767e5,
+      #12bcfe,
+      #44ce7b,
+      #4fcf70,
+      #fad648
+    );
+  }
+
+  100% {
+    border-image-source: linear-gradient(
```
Comment (shahednasser) [inline src/components/special/_rainbow-border.less:71]: Change this to 
```
background-position: 0 0, 0 50%;
```

**Review by shahednasser — CHANGES_REQUESTED** [review]
So the problem is that you're using an image for the border. Instead, use the following:

```
    background-image: linear-gradient(#000, #000),linear-gradient( 115deg, #4fcf70, #fad648, #a767e5, #12bcfe, #44ce7b );
    background-origin: padding-box, border-box;
    background-repeat: no-repeat;
    border: 5px solid transparent;
    background-position: 0 0, 0% 50%;
    background-size: 100% 100%, 200% 200%;
```

And make the changes as requested below.
If you wanna play around with the animation and change it that's fine too.

**Comment by Soumi150** [comment]
Thank you very much.

File: src/components/special/_rainbow-border.less [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  background-image: linear-gradient(#000, #000),
+    linear-gradient(115deg, #4fcf70, #fad648, #a767e5, #12bcfe, #44ce7b);
+  background-origin: padding-box, border-box;
+  background-repeat: no-repeat;
+  border: 5px solid transparent;
+  background-position: 0 0, 0% 50%;
+  background-size: 100% 100%, 200% 200%;
+
+  &:hover {
+    animation: colorwave 1.6s ease infinite;
```
Comment (shahednasser) [inline src/components/special/_rainbow-border.less:11]: I think it would look better if it was slower so maybe change it to 3.6s

**Review by shahednasser — CHANGES_REQUESTED** [review]
Looks good but in the animation you added the 0% and 50% but not the 100% so please fix that

**Comment by shahednasser** [comment]
Great work thank you!

@all-contributors please add @Soumi150 for code

**Pull request merged**